### PR TITLE
Add support for cross-domain catch-all redirects + implement VAMC redirects

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
**Issues:**
https://github.com/department-of-veterans-affairs/va.gov-team/issues/26199
https://github.com/department-of-veterans-affairs/va.gov-team/issues/26234
https://github.com/department-of-veterans-affairs/va.gov-team/issues/26200

This PR adds support for catch-all cross-domain redirects + implements the following redirects:

Current URL  |  Redirect Destination or New URL
---  |  ---
https://www.altoona.va.gov/  <br>  **NOTE:  This redirect is a catch-all for the top level of this sub-domain AND all remaining pages in the sub-domain that are not specified as a 1:1 redirect in the list below.**  | https://www.va.gov/altoona-health-care/
https://www.altoona.va.gov/locations/directions.asp | https://www.va.gov/altoona-health-care/locations/james-e-van-zandt-veterans-administration-medical-center/
https://www.altoona.va.gov/locations/DuBois.asp | https://www.va.gov/altoona-health-care/locations/dubois-va-clinic/
https://www.altoona.va.gov/locations/Huntingdon_VA_Community_Based_Outpatient_Clinic.asp | https://www.va.gov/altoona-health-care/locations/huntingdon-county-va-clinic/
https://www.altoona.va.gov/locations/Indiana_VA_Outpatient_Clinic.asp | https://www.va.gov/altoona-health-care/locations/indiana-county-va-clinic/
https://www.altoona.va.gov/locations/Johnstown.asp | https://www.va.gov/altoona-health-care/locations/johnstown-va-clinic/
https://www.altoona.va.gov/locations/StateCollege.asp | https://www.va.gov/altoona-health-care/locations/state-college-va-clinic/
https://www.altoona.va.gov/services/covid-19-vaccines.asp | **Modified by MN** <br> ~https://www.va.gov/altoona-health-care/health-services/covid-19-vaccines/~ <br> https://www.va.gov/altoona-health-care/programs/covid-19-vaccines/

Current URL  |  Redirect Destination or New URL
---  |  ---
https://www.wilkes-barre.va.gov/  <br>  **NOTE:  This redirect is a catch-all for the top level of this sub-domain AND all remaining pages in the sub-domain that are not specified as a 1:1 redirect in the list below.**   | https://www.va.gov/wilkes-barre-health-care/
https://www.wilkes-barre.va.gov/locations/directions.asp | https://www.va.gov/wilkes-barre-health-care/locations/wilkes-barre-va-medical-center/ 
https://www.wilkes-barre.va.gov/locations/Allentown.asp | https://www.va.gov/wilkes-barre-health-care/locations/allentown-va-clinic/
https://www.wilkes-barre.va.gov/locations/Berwick.asp | https://www.va.gov/wilkes-barre-health-care/locations/columbia-county-va-clinic/
https://www.wilkes-barre.va.gov/locations/Bangor.asp | https://www.va.gov/wilkes-barre-health-care/locations/northampton-county-va-clinic/
https://www.wilkes-barre.va.gov/locations/Sayre.asp | https://www.va.gov/wilkes-barre-health-care/locations/sayre-va-clinic/
https://www.wilkes-barre.va.gov/locations/Tobyhanna.asp |   https://www.va.gov/wilkes-barre-health-care/locations/tobyhanna-va-clinic/
https://www.wilkes-barre.va.gov/locations/Honesdale_Wayne_County.asp | https://www.va.gov/wilkes-barre-health-care/locations/wayne-county-va-clinic/
https://www.wilkes-barre.va.gov/locations/Williamsport.asp | https://www.va.gov/wilkes-barre-health-care/locations/williamsport-va-clinic/
https://www.wilkes-barre.va.gov/services/covid-19-vaccines.asp | https://www.va.gov/wilkes-barre-health-care/programs/covid-19-vaccines/

Current URL  |  Redirect Destination or New URL
---  |  ---
https://www.wilmington.va.gov/ 		<br>  **NOTE:  This redirect is a catch-all for the top level of this sub-domain AND all remaining pages in the sub-domain that are not specified as a 1:1 redirect in the list below.**		| https://www.va.gov/wilmington-health-care/
https://www.wilmington.va.gov/locations/directions.asp 	| https://www.va.gov/wilmington-health-care/locations/wilmington-va-medical-center/
https://www.wilmington.va.gov/locations/Atlantic_County_Community_Based_Outpatient_Clinic.asp | https://www.va.gov/wilmington-health-care/locations/atlantic-county-va-clinic/
https://www.wilmington.va.gov/locations/Cape_May_County_Community_Based_Outpatient_Clinic.asp | https://www.va.gov/wilmington-health-care/locations/cape-may-county-va-clinic/
https://www.wilmington.va.gov/locations/Cumberland_County_Community_Based_Outpatient_Clinic.asp | https://www.va.gov/wilmington-health-care/locations/cumberland-county-va-clinic/
https://www.wilmington.va.gov/locations/Kent_County_Community_Based_Outpatient_Clinic.asp | https://www.va.gov/wilmington-health-care/locations/kent-county-va-clinic/
https://www.wilmington.va.gov/locations/Sussex_County_Community_Based_Outpatient_Clinic.asp | https://www.va.gov/wilmington-health-care/locations/sussex-county-va-clinic/
https://www.wilmington.va.gov/services/covid-19-vaccines.asp | ~~https://www.va.gov/wilmington-health-care/health-services/covid-19-vaccines/~~ ~New: https://www.va.gov/wilmington-health-care/covid-19-vaccines/~ <br> **UPDATED by MN** https://www.va.gov/wilmington-health-care/programs/covid-19-vaccines/

## Testing done
Updated unit tests to support cross-domain catch-all redirects.

## Screenshots
N/A

## Acceptance criteria
- [x] adds support for catch-all cross-domain redirects + implements the above redirects

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by zacharymorel — APPROVED** [review]

File: src/applications/proxy-rewrite/redirects/crossDomainRedirects.json [hunk 1]
```diff
@@ -1098,5 +1098,46 @@
     "domain": "www.benefits.va.gov",
     "src": "/gibill/vrrap.asp",
     "dest": "/education/other-va-education-benefits/veteran-rapid-retraining-assistance/"
+  },
+  {
+    "domain": "www.altoona.va.gov",
+    "src": "/",
+    "dest": "/altoona-health-care/",
+    "catchAll": true
```
Comment (zacharymorel) [inline src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1106]: Is there any existing redirects that act as catch all or is this a first of it's kind?

File: src/applications/proxy-rewrite/redirects/index.js [hunk 2]
```diff
@@ -2,23 +2,56 @@ import environment from 'platform/utilities/environment';
...
     // eslint-disable-next-line no-param-reassign
     currentWindow.location.href = `${environment.BASE_URL}${
-      matchedRedirect.dest
+      catchAllRedirectMatch.dest
     }`;
+    return;
   }
 }
+
+export default redirectIfNecessary;
```
Comment (zacharymorel) [inline src/applications/proxy-rewrite/redirects/index.js:57]: The entire file has great variable names. Nice work!

File: src/applications/proxy-rewrite/redirects/crossDomainRedirects.json (hunk 1 above)
Comment (kelsonic) [reply src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1106]: First of its kind unfortunately, this PR scares me quite a bit lol

File: src/applications/proxy-rewrite/redirects/index.js (hunk 2 above)
Comment (kelsonic) [reply src/applications/proxy-rewrite/redirects/index.js:57]: Thanks so much! 🙂

File: src/applications/proxy-rewrite/redirects/crossDomainRedirects.json (hunk 1 above)
Comment (zacharymorel) [reply src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1106]: Yeah, this is one of those like, okay, _we've tested locally, now we gotta push to see it live_...

Comment (k80bowman) [reply src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1106]: This one scares me, too. Would it be better to merge after the daily deploy goes out so we have a good chunk of time to test before this goes to prod? Do you have a plan for testing this to make sure things are OK before this change hits prod?

Comment (kelsonic) [reply src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1106]: I won't be merging this until tomorrow. I'm not entirely sure how to test this fully before it goes to production tbh 🤔 I'm trying to think of ways currently

Comment (kelsonic) [reply src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1106]: Hey @k80bowman I added unit tests that fully test catch-all redirects, so feeling much more confident about this now! 🔥

Comment (k80bowman) [reply src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1106]: Fantastic! 🏆

**Review by k80bowman — APPROVED** [review]

File: src/applications/proxy-rewrite/tests/redirects/index.unit.spec.js [hunk 3]
```diff
@@ -29,6 +30,37 @@ describe('Redirect replaced pages', () => {
...
+        host: 'www.altoona.va.gov',
+        pathname: '/blahblahblah',
+      },
+    };
+
+    redirectIfNecessary(fakeWindow);
+
+    expect(fakeWindow.location.href.endsWith('.gov/altoona-health-care/')).to.be
+      .true;
+  });
```
Comment (zacharymorel) [inline src/applications/proxy-rewrite/tests/redirects/index.unit.spec.js:63]: Nice work!

File: src/applications/proxy-rewrite/redirects/crossDomainRedirects.json [hunk 4]
```diff
@@ -1098,5 +1098,138 @@
...
+  },
+  {
+    "domain": "www.altoona.va.gov",
+    "src": "/locations/directions.asp",
+    "dest": "/altoona-health-care/locations/james-e-van-zandt-veterans-administration-medical-center/"
```
Comment (mnorthuis) [inline src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1111]: Let's remove this 1:1 redirect, we do not need to redirect the legacy directions page.

File: src/applications/proxy-rewrite/redirects/crossDomainRedirects.json [hunk 5]
```diff
@@ -1098,5 +1098,138 @@
...
+  {
+    "domain": "www.wilmington.va.gov",
+    "src": "/",
+    "dest": "/wilmington-health-care/",
+    "catchAll": true
+  },
+  {
+    "domain": "www.wilmington.va.gov",
+    "src": "/locations/directions.asp",
+    "dest": "/wilmington-health-care/locations/wilmington-va-medical-center/"
```
Comment (mnorthuis) [inline src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1152]: Let's remove this 1:1 redirect, we do not need to redirect the legacy directions page.

File: src/applications/proxy-rewrite/redirects/crossDomainRedirects.json [hunk 6]
```diff
@@ -1098,5 +1098,138 @@
...
+  {
+    "domain": "www.wilkes-barre.va.gov",
+    "src": "/",
+    "dest": "/wilkes-barre-health-care/",
+    "catchAll": true
+  },
+  {
+    "domain": "www.wilkes-barre.va.gov",
+    "src": "/locations/directions.asp",
+    "dest": "/wilkes-barre-health-care/locations/wilkes-barre-va-medical-center/"
```
Comment (mnorthuis) [inline src/applications/proxy-rewrite/redirects/crossDomainRedirects.json:1193]: Let's remove this 1:1 redirect, we do not need to redirect the legacy directions page.

File: src/applications/proxy-rewrite/tests/redirects/index.unit.spec.js [hunk 7]
```diff
@@ -29,6 +30,37 @@ describe('Redirect replaced pages', () => {
 
     expect(fakeWindow.location.href).to.be.undefined;
   });
+
+  it('should redirect absolute redirects over catch-all redirects', () => {
+    const fakeWindow = {
+      location: {
+        host: 'www.altoona.va.gov',
+        pathname: '/locations/directions.asp',
```
Comment (mnorthuis) [inline src/applications/proxy-rewrite/tests/redirects/index.unit.spec.js:38]: Let's remove this 1:1 redirect, we do not need to redirect the legacy directions page.

**Pull request merged**